doc/rtt.js

```diff
@@ -0,0 +1,178 @@
+ * @property {String} log The complete log from the internal functions.
+ *
+ * @example
+ * RTT0.start(12345678, {}, function(err, down, up){
+ *     if (err) {
+ *         throw err;
+ *     } else {
+ *         // success
+ *     }
+ * });
+ *
+ * @example
+ * RTT.start(12345678, {}, function(err, down, up){
+ *     if (err && err.errno === nrfjprogjs.RTTCouldNotFindControlBlock) {
+ *          console.error('The firmware is not RTT capable');
+ *     }
+ * });
+ */
+
+/**
+ * Information about the different up and down channels available on the device
+ * @typedef ChannelInfo
+ * @property {integer} channelIndex The index used to address this channel
+ * @property {String} name The name of the channel
+ * @property {integer} size The size of the channel
+ */
+
+ /**
+ * Option flags to be used when starting RTT. This may speed up the process of locating the control block, and the RTT Start.
+ * @typedef StartOptions
+ * @property {integer} controlBlockLocation The location of the control block. If this location is not the start of the RTT control block, start will fail.
+ */
+
+
+/**
+ * Async function to start RTT
+ *
+ * @example
+ * RTT.start(12345678, {}, function(err, down, up){
+ *      if (err) console.error('The firmware is not RTT capable');
+ *
+ *      console.log('Down channels');
+ *      down.map(function(channel) {
+ *             console.log('Index:', channel.channelIndex);
+ *             console.log('Name:', channel.name);
+ *             console.log('Size:', channel.size);
+ *      });
+ *
+ *      console.log('Up channels');
+ *      up.map(function(channel) {
+ *             console.log('Index:', channel.channelIndex);
+ *             console.log('Name:', channel.name);
+ *             console.log('Size:', channel.size);
+ *      });
+ * });
+ *
+ * @param {integer} serialNumber The serial number of the device to start RTT on
+ * @param {integer} startOptions A plain object containing options about how to start RTT
+ * @param {Function} callback A callback function to handle the async response.
+ *   It shall expect three parameters: ({@link module:pc-nrfjprog-js~Error|Error}, Array of {@link module:pc-nrfjprog-js~Version|ChannelInfo},
+ *   Array of {@link module:pc-nrfjprog-js~Version|ChannelInfo})
+ */
+export function start(serialnumber, startoptions, callback) {}
+
+/**
+ * Async function to stop RTT
+ *
+ * @example
+ * RTT.stop(function(err){
+ *     if (err) console.error('Stopping RTT failed');
+ * });
+ *
+ * @param {Function} callback A callback function to handle the async response.
+ *   It shall expect one parameter: ({@link module:pc-nrfjprog-js~Error|Error})
+ */
+export function stop(callback) {}
+
+/**
+ * Async function to read RTT contents on a device.
+ *
+ * @example
+ * RTT.read(0, 100, function(err, stringData, rawData, timeSinceRTTStartInUs) {
+ *      if (err) throw err;
+ *      console.log(stringData);
+ *      console.log(rawData);
```

Comment:
Maybe we should write about how this looks similar to an `Uint8Array` but not quite. 🤔

Reply:
I will have to look into if `Uint8Array` is a better option here. If it is compatible with normal arrays, I may change the implementation.